components/dot_menu/dot_menu.jsx

```diff
@@ -303,6 +313,12 @@ export default class DotMenu extends Component {
                         text={Utils.localizeMessage('post_info.pin', 'Pin')}
                         onClick={this.handlePinMenuItemActivated}
                     />
+                    <MenuItemAction
+                        id={`unread_post_${this.props.post.id}`}
+                        show={!isSystemMessage}
```

Comment:
Is this not supposed to be visible for system messages? I thought the spec said you could mark a system message as unread.

If you can actually mark a system message as unread though, we might have some extra work to do though. The client combines messages where users have joined/left a channel around the same time, and since those posts don't actually exist on the server, we'll have to make sure that you can mark those posts as unread properly

Reply:
I thought ephemeral messages where system messages, am I getting it wrong?
I got the idea from no. 4 in https://projects.invisionapp.com/share/N4RBXGMY9QR#/screens/355357487
I also just realized that I'm allowing to unread from RHS